src/main/java/org/reaktivity/nukleus/http2/internal/types/stream/Http2ContinuationFW.java

```diff
@@ -0,0 +1,169 @@
+        length += (buffer().getByte(offset() + LENGTH_OFFSET + 1) & 0xFF) << 8;
+        length += buffer().getByte(offset() + LENGTH_OFFSET + 2) & 0xFF;
+        return length;
+    }
+
+    public Http2FrameType type()
+    {
+        //assert buffer().getByte(offset() + TYPE_OFFSET) == CONTINUATION.getType();
+        return CONTINUATION;
+    }
+
+    public byte flags()
+    {
+        return buffer().getByte(offset() + FLAGS_OFFSET);
+    }
+
+    // streamId != 0, caller to validate
+    public int streamId()
+    {
+        // Most significant bit is reserved and is ignored when receiving
+        return buffer().getInt(offset() + STREAM_ID_OFFSET, BIG_ENDIAN) & 0x7F_FF_FF_FF;
+    }
+
+    public boolean endHeaders()
+    {
+        return Http2Flags.endHeaders(flags());
+    }
+
+    public void forEach(Consumer<HpackHeaderFieldFW> headerField)
+    {
+        headerBlockRO.forEach(headerField);
+    }
+
+    @Override
+    public int limit()
+    {
+        return offset() + PAYLOAD_OFFSET + payloadLength();
+    }
+
+    @Override
+    public Http2ContinuationFW wrap(DirectBuffer buffer, int offset, int maxLimit)
+    {
+        super.wrap(buffer, offset, maxLimit);
+        headerBlockRO.wrap(buffer(), offset() + PAYLOAD_OFFSET, offset() + PAYLOAD_OFFSET + payloadLength());
+
+        checkLimit(limit(), maxLimit);
+        return this;
+    }
+
+    @Override
+    public String toString()
+    {
+        return String.format("%s frame <length=%s, type=%s, flags=%s, id=%s>",
+                type(), payloadLength(), type(), flags(), streamId());
+    }
+
+    public static final class Builder extends Flyweight.Builder<Http2ContinuationFW>
+    {
+        private final HpackHeaderBlockFW.Builder blockRW = new HpackHeaderBlockFW.Builder();
+
+        public Builder()
+        {
+            super(new Http2ContinuationFW());
+        }
+
+        @Override
+        public Builder wrap(MutableDirectBuffer buffer, int offset, int maxLimit)
+        {
+            super.wrap(buffer, offset, maxLimit);
+
+            int length = 0;
+            buffer().putByte(offset() + LENGTH_OFFSET, (byte) ((length & 0x00_FF_00_00) >>> 16));
+            buffer().putByte(offset() + LENGTH_OFFSET +1, (byte) ((length & 0x00_00_FF_00) >>> 8));
+            buffer().putByte(offset() + LENGTH_OFFSET + 2, (byte) ((length & 0x00_00_00_FF)));
+
+            buffer().putByte(offset() + TYPE_OFFSET, CONTINUATION.getType());
+
+            buffer().putByte(offset() + FLAGS_OFFSET, (byte) 0);
+
+            limit(offset() + PAYLOAD_OFFSET);
+
+            blockRW.wrap(buffer(), offset() + PAYLOAD_OFFSET, maxLimit());
+
+            return this;
+        }
+
+        public Builder streamId(int streamId)
+        {
+            buffer().putInt(offset() + STREAM_ID_OFFSET, streamId, ByteOrder.BIG_ENDIAN);
+            return this;
+        }
+
+        public Builder endHeaders()
+        {
```

Comment:
I think it is not as nice as builder.endHeaders() or builder.endHeaders(true), right ?